pkgs/stdenv/generic/setup.sh

```diff
@@ -331,6 +331,17 @@ isELF() {
     if [ "$magic" = $'\177ELF' ]; then return 0; else return 1; fi
 }
 
+# Return success if the specified file is a .a archive.
+isStaticLib() {
+    local fn="$1"
+    local fd
+    local magic
+    exec {fd}< "$fn"
+    read -r -N 8 -u "$fd" magic
+    exec {fd}<&-
+    if [ "$magic" = $'!<arch>\n' ]; then return 0; else return 1; fi
+}
+
```

Comment:
Need for these helpers feels like a workaround. I suggest avoiding them if possible.